solidity/contracts/libs/hooks/GlobalHookMetadata.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+/**
+ * Format of metadata:
+ *
+ * [0:1] variant
+ * [1:33] msg.value
+ * [34:65] Gas limit for message (IGP)
+ * [66:85] Refund address for message (IGP)
+ * [86:] Custom metadata
+ */
+library GlobalHookMetadata {
+    uint8 private constant VARIANT_OFFSET = 0;
+    uint8 private constant MSG_VALUE_OFFSET = 1;
+    uint8 private constant GAS_LIMIT_OFFSET = 33;
+    uint8 private constant REFUND_ADDRESS_OFFSET = 65;
+    uint256 private constant MIN_METADATA_LENGTH = 85;
```

Comment:
I personally would only advocate for msg value to be global out of these 3

Reply:
cc @nambrot

Reply:
I would make all three global as well just because these are the 3 parameters we do expect people to use (as well as custom metadata to be appended after).